src/h5web/visualizations/heatmap/HeatmapVis.tsx

```diff
@@ -26,31 +27,20 @@ function HeatmapVis(): JSX.Element {
 
   return (
     <div className={styles.root}>
-      <div ref={mapAreaRef} className={styles.mapArea}>
-        {heatmapStyles && (
-          <div className={styles.heatmap} style={heatmapStyles}>
-            <Canvas
-              className={styles.canvasWrapper}
-              orthographic
-              invalidateFrameloop
-            >
-              <ambientLight />
-              {/* Provide context again - https://github.com/react-spring/react-three-fiber/issues/262 */}
-              <HeatmapProvider {...props}>
-                <AxisGrid
-                  // -0.5 to have ticks at the center of pixels
-                  abscissaDomain={[-0.5, cols - 0.5]}
-                  ordinateDomain={[-0.5, rows - 0.5]}
-                  axisOffsets={axisOffsets}
-                />
-                <TooltipMesh dims={dims} data={data} />
-                <PanZoomMesh />
-                <Mesh />
-              </HeatmapProvider>
-            </Canvas>
-          </div>
-        )}
-      </div>
+      <VisCanvas axisOffsets={axisOffsets} aspectRatio={aspectRatio}>
+        {/* Provide context again - https://github.com/react-spring/react-three-fiber/issues/262 */}
+        <HeatmapProvider {...props}>
+          <AxisGrid
+            // -0.5 to have ticks at the center of pixels
+            abscissaDomain={[-0.5, cols - 0.5]}
+            ordinateDomain={[-0.5, rows - 0.5]}
+            axisOffsets={axisOffsets}
+          />
```

Comment:
`axisOffsets` are passed as props for `VisCanvas` and `AxisGrid`. You thought about a provider but it might not be worth it for only two components ?

Reply:
I think you had convinced me that `AxisGrid` could be rendered by `VisCanvas` since it doesn't need to be inside the `HeatmapProvider` anymore. 😅 I wasn't a fan of forwarding the domain props, but you're right, it's not as bad as passing `axisOffsets` in two places.